### ISSUE TITLE
Type issue with ActionCreators from createSlice with no payload

### ISSUE THREAD

**Issue by pat-son** [reduxjs/redux-starter-kit#108]
I am creating a slice for my state, and one of the actions I want to create does not take a payload. However, when mapping the dispatch to the props in `connect`, I get a weird type error.
![image](https://user-images.githubusercontent.com/8462638/52240423-c9608300-289e-11e9-8a7f-cb5bf13fcc0f.png)
`
Argument of type 'typeof MyComponent' is not assignable to parameter of type 'ComponentType<Matching<{ loading: boolean; } & { setLoaded: SliceActionCreator<void>; }, MyProps>>'.
  Type 'typeof MyComponent' is not assignable to type 'ComponentClass<Matching<{ loading: boolean; } & { setLoaded: SliceActionCreator<void>; }, MyProps>, any>'.
    Types of parameters 'props' and 'props' are incompatible.
      Type 'Matching<{ loading: boolean; } & { setLoaded: SliceActionCreator<void>; }, MyProps>' is not assignable to type 'Readonly<MyProps>'.
        Types of property 'setLoaded' are incompatible.
          Type 'SliceActionCreator<void>' is not assignable to type '() => void'.
`

Here's a minimal reproduction of the issue I'm having:

```Typescript
import React, { PureComponent } from 'react';
import { createSlice } from 'redux-starter-kit';
import { connect } from 'react-redux'

interface MyState {
    loading: boolean;
}

const INITIAL_STATE: MyState = {
    loading: true,
}

const slice = createSlice({
    slice: 'mySlice',
    initialState: INITIAL_STATE,
    reducers: {
        setLoaded: (state) => { state.loading = false; },
    }
})

interface MyProps {
    loading: boolean;
    setLoaded(): void;
}

class MyComponent extends PureComponent<MyProps> {
    componentDidMount() {
        // perform some check for loading
        const setLoaded = this.props.setLoaded;

        setLoaded();
    }

    render() {
        const loading = this.props.loading;
        if (loading) {
            return (
                <p>Loading...</p>
            );
        } else {
            return (
                <p>Loaded!</p>
            )
        }
    }
}

export default connect(
    (state: MyState) => ({
        loading: state.loading,
    }),
    {
        setLoaded: slice.actions.setLoaded,
    }
)(MyComponent)

```

If I change the type of the action property in `MyProps` to take a void payload, then the error goes away, but that's obviously not the correct type:

```Typescript
interface MyProps {
    loading: boolean;
    setLoaded(payload: void): void;
}
```

Am I doing something wrong, or is there a bug in the typing?

**Comment by denisw** [comment]
This is indeed a typing bug. I created a PR (#110) which fixes the issue.

**Comment by pat-son** [comment]
Awesome, thanks

**Comment by pat-son** [comment]
Fixed by #133.
